src/backend/joanie/core/models/products.py

```diff
@@ -591,6 +591,33 @@ def main_invoice(self) -> dict | None:
         except ObjectDoesNotExist:
             return None
 
+    @property
+    def is_free(self):
+        """
+        Return True if the order is free.
+        """
+        return not self.total
+
+    @property
+    def has_payment_method(self):
+        """
+        Return True if the order has a payment method.
+        """
+        return self.owner.credit_cards.filter(
+            is_main=True,
+            initial_issuer_transaction_identifier__isnull=False,
+        ).exists()
```

Comment:
I'm afraid that we have to link a credit card to an order otherwise once a user will have save a payment method, it will not be able to use a new one...

Reply:
I suppose we will encounter this soon…